### PR TITLE
Bump eventing 1.15

### PR DESCRIPTION
**Pull request by pierDipi**
- Bump eventing
- Bump deps (as EKB tests needs updated code)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift-knative/serverless-operator
 
-go 1.21.0
+go 1.22
 
 toolchain go1.22.5
```
Comment (dsimansk) [inline go.mod:5]: Could you pls drop `toolchain` while it's already changing `go.mod` go version? Thanks! :)

Comment (pierDipi) [reply go.mod]: Done

**Comment by pierDipi** [comment]
```

go: go.mod requires go >= 1.22 (running go 1.21.11; GOTOOLCHAIN=local)
error: build error: building at STEP "RUN go build -o /tmp/metadata-webhook github.com/openshift-knative/serverless-operator/serving/metadata-webhook/cmd/webhook": while running runtime: exit status 1
```

**Comment by pierDipi** [comment]
```
{Failed  === RUN   TestTraceStartedAtActivator
W0913 15:36:37.952330   45028 warnings.go:70] Kubernetes default value is insecure, Knative may default this to secure in a future release: spec.template.spec.containers[0].securityContext.allowPrivilegeEscalation, spec.template.spec.containers[0].securityContext.capabilities, spec.template.spec.containers[0].securityContext.runAsNonRoot, spec.template.spec.containers[0].securityContext.seccompProfile
    spoof.go:111: Spoofing testtracestartedatactivator-serverless-tests.apps.serverless-ocp-4-15-amd64-aws-us-east-1-cspc2.serverless.devcluster.openshift.com -> testtracestartedatactivator-serverless-tests.apps.serverless-ocp-4-15-amd64-aws-us-east-1-cspc2.serverless.devcluster.openshift.com
    spoof.go:111: Spoofing testtracestartedatactivator-serverless-tests.apps.serverless-ocp-4-15-amd64-aws-us-east-1-cspc2.serverless.devcluster.openshift.com -> testtracestartedatactivator-serverless-tests.apps.serverless-ocp-4-15-amd64-aws-us-east-1-cspc2.serverless.devcluster.openshift.com
    tracing_test.go:272: service: activator-service
    tracing_test.go:272: service: custom-route-test-00001-deployment-5ff56c6f76-wvhlv
    tracing_test.go:272: service: testtracestartedatactivator-00001-deployment-55d6974575-rlrht
    tracing_test.go:272: service: activator-service
    tracing_test.go:272: service: custom-route-test-00001-deployment-5ff56c6f76-wvhlv
    tracing_test.go:272: service: testtracestartedatactivator-00001-deployment-55d6974575-rlrht
    tracing_test.go:272: service: activator-service
    tracing_test.go:272: service: custom-route-test-00001-deployment-5ff56c6f76-wvhlv
    tracing_test.go:272: service: testtracestartedatactivator-00001-deployment-55d6974575-rlrht
    tracing_test.go:272: service: activator-service
    tracing_test.go:272: service: custom-route-test-00001-deployment-5ff56c6f76-wvhlv
    tracing_test.go:272: service: testtracestartedatactivator-00001-deployment-55d6974575-rlrht
    tracing_test.go:87: Error verifying traces: context deadline exceeded: error getting services: rpc error: code = Internal desc = failed to fetch services: plugin error: rpc error: code = Unknown desc = failed GET to tempo: Get "http://127.0.0.1:3200/api/search/tag/service.name/values?start=1725982637&end=1726241837": dial tcp 127.0.0.1:3200: connect: connection refused
--- FAIL: TestTraceStartedAtActivator (40.56s)
}
```
---

```
{  pods/knative-eventing_kafka-controller-655886447-f7lj4_controller_previous.log.gz:E0913 15:35:38.574598       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)}
```

**Comment by pierDipi** [comment]
Fix for the panic issue https://github.com/openshift-knative/eventing-kafka-broker/pull/1256

**Comment by pierDipi** [comment]
```
"TestServerlessUpgradePrePost/PreUpgradeTests/InMemoryChannelPreUpgradeTest"}
2024-09-13T15:30:59.597Z	INFO	test	injection/injection.go:63	Starting informers...
W0913 15:30:59.607522   47449 reflector.go:539] knative.dev/pkg/controller/controller.go:726: failed to list *v1alpha1.EventPolicy: the server could not find the requested resource (get eventpolicies.eventing.knative.dev)
```

The test running is trying to setup an informer for authz resources ... looking

**Comment by pierDipi** [comment]
Another PR: https://github.com/openshift-knative/eventing-kafka-broker/pull/1257

**Comment by pierDipi** [comment]
```
error getting cluster operators: Get "https://api.serverless-ocp-4-15-amd64-aws-us-east-1-x28g4.serverless.devcluster.openshift.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 98.82.197.251:6443: connect: connection refused
```

**Comment by pierDipi** [comment]
```
2024-09-18T11:08:23.580Z	DEBUG	test	milestone/emitter_log.go:57	Environment {
  "featureState": "Any",
  "namespace": "test-hwqhmvgg",
  "requirementLevel": "All"
 }	{"namespace": "test-hwqhmvgg"}
    triggerv2_downgrade.go:56: failed to downgrade from triggerv2 context deadline exceeded
--- FAIL: TestServerlessUpgradePrePost/Run/Steps/DowngradeWith/DowngradeServerless (284.97s)
}
```
Making progress ...

**Comment by pierDipi** [comment]
```
   verify.go:74: Error fetching complete/inactive report: context deadline exceeded
        Report: &{State:active EventsSent:7525 TotalRequests:0 Thrown:{Duplicated:[event #7375 received 2 times, but should be received only once event #7353 received 2 times, but should be received only once event #7369 received 2 times, but should be received only once event #7395 received 2 times, but should be received only once event #7402 received 2 times, but should be received only once] Missing:[] Unexpected:[] Unavailable:[]}}
--- FAIL: TestServerlessUpgradeContinual/Run/BrokerBackedByChannelContinualTest (983.63s)
```

**Comment by pierDipi** [comment]
@matzew @skonto this is looking good now

**Comment by pierDipi** [comment]
I'll let the bot do the generated files after we merge this

**Comment by skonto** [comment]
I guess we can fix lint issues in another PR.

File: vendor/knative.dev/operator/pkg/reconciler/common/workload_override.go [hunk 2]
```diff
@@ -223,7 +223,7 @@ func replaceProbes(override *base.WorkloadOverride, ps *corev1.PodTemplateSpec)
 				}
 				if *overrideProbe == (v1.Probe{}) {
 					//  Disable probe when users explicitly set the empty overrideProbe.
-					containers[i].ReadinessProbe = nil
+					containers[i].LivenessProbe = nil
```
Comment (skonto) [inline vendor/knative.dev/operator/pkg/reconciler/common/workload_override.go:226]: Probably we should have backported this https://github.com/knative/operator/pull/1823

Comment (matzew) [reply vendor/knative.dev/operator/pkg/reconciler/common/workload_override.go:226]: we could still, but than... the next "update dependency" has the change

Comment (pierDipi) [reply vendor/knative.dev/operator/pkg/reconciler/common/workload_override.go:226]: I guess you mean in 1.34 ?

**Pull request merged**